### PR TITLE
Fix bad access crash for fetchOlderTransactions #6251

### LINKED ISSUE #6251: Fix bas access crash for fetchOlderTransactions

### PR DESCRIPTION
**Pull request by oa-s**
Closes #6251

### CODE REVIEW

File: modules/AlphaWalletFoundation/AlphaWalletFoundation/Transactions/EtherscanSingleChainTransactionProvider.swift [hunk 1]
```diff
@@ -184,7 +183,7 @@ class EtherscanSingleChainTransactionProvider: SingleChainTransactionProvider {
 
         transactionsNetworkProvider
             .getTransactions(startBlock: 1, endBlock: oldestCachedTransaction.blockNumber - 1, sortOrder: .desc)
-            .sink(receiveCompletion: { [transactionsTracker] result in
+            .sinkAsync(receiveCompletion: { [transactionsTracker] result in
```
Comment (hboon) [inline modules/AlphaWalletFoundation/AlphaWalletFoundation/Transactions/EtherscanSingleChainTransactionProvider.swift:186]: Wouldn't it be the same?

Comment (oa-s) [reply modules/AlphaWalletFoundation/AlphaWalletFoundation/Transactions/EtherscanSingleChainTransactionProvider.swift:186]: there is an issue with accessing value type types like arrays, sets, dictionaries in background, updating it with `sinkAsync` avoids this issue as we don't store it in any array, and just release subscription when receive completion

Comment (hboon) [reply modules/AlphaWalletFoundation/AlphaWalletFoundation/Transactions/EtherscanSingleChainTransactionProvider.swift:186]: Ah I see.

**Review by hboon — APPROVED** [review]

**Comment by hboon** [comment]
"bas" typo here too

**Pull request merged**